### ISSUE TITLE
Check Sagemaker costs

### ISSUE THREAD

**Issue by michaelgloeckner** [onecx-apps/onecx-chat#6]
Check sagemaker costs
Check if we can scale it up and down to only be used on business hours.

**Comment by lmitlaender** [comment]
### Pricing:
Instance pricing is around 25 - 50% more expensive for sagemaker ml instances compared to ec2 instances of the same type. The percentage seems to be higher for lower cost instances and lower for higher cost ones.

Comparing the Hourly Rate at the example of g4dn instances:
Sagemaker:
![image](https://github.com/onecx-apps/onecx-chat/assets/24322646/cff92608-d42c-4138-98d4-d2e364e7f4aa)

EC2:
![image](https://github.com/onecx-apps/onecx-chat/assets/24322646/014449c6-fef7-41e9-a552-a0d4b2b5ca14)

Data Prices for Sagemaker are 0,016 USD per GB for Data transfer IN and OUT
Data Prices for EC2 are free inbound traffic and around 0,05 to 0,09 USD per GB for Outbound traffic

Sagemaker does reduce the need for a Kubernetes Cluster, but is far less flexible regarding where it can be hosted since it is AWS proprietary. It does however give many QoL Tools for integrating with other AWS Services as a managed service.

### Scaling:
Sagemaker sadly doesn't support auto scaling to zero, the minimum instance count for an endpoint always has to be at least one. Otherwise its scaling options are similar to ec2 Auto Scaling.

### TL;DR:
Sagemaker can be worth it for very high outbound data usage endpoints, and especially when trying to set up more training related jobs or a full Machine Learning Pipeline from Data to Training to Evaluation to Deployment.

For our use-case of only hosting an endpoint the only really interesting part could be the ability of quickly deploying jumpstart models to test different models.
